liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/applications/InstallAppMojoSupport.java

```diff
@@ -139,6 +140,15 @@ protected void installLooseConfigApp() throws Exception {
         config.toXmlFile(looseConfigFile);
     }
     
+    private boolean containsJavaSource(MavenProject proj) {       
+        File javaSourceDir = new File(proj.getBasedir(), "src/main/java");
+        if (javaSourceDir.exists() && javaSourceDir.isDirectory() && javaSourceDir.listFiles().length > 0) {
```

Comment:
If this directory just contains a folder, it will return true regardless if it contains class files or not.